qtt/algorithms/markov_chain.py

```diff
@@ -0,0 +1,241 @@
+# -*- coding: utf-8 -*-
+""" Class to generate signals with continous-time Markov chains
+
+
+@author: pieter.eendebak@gmail.com
+"""
+
+#%%
+import numpy as np
+import random
+import scipy.linalg
+import itertools
+
+def _solve_least_squares(A, B):
+    from distutils.version import StrictVersion
+    if StrictVersion(np.__version__) < StrictVersion('1.14.0'):
+        # legacy numpy versions
+        rcond=-1
+    else:
+        rcond = None
+    solution = np.linalg.lstsq(A, B, rcond=rcond)[0]
+    return solution
+
+class ContinuousTimeMarkovModel:
+
+    def __init__(self, states, holding_parameters, jump_chain):
+        """ Class that represents a continous-time Markov chain
+
+        Args:
+            states (str[]): list with names for the states
+            holding_parameters(float[]): list with the holding parameters
+            jump_chain (array): The jump chain or transition matrix
+
```

Comment:
Is it possible to very shortly explain what holding parameters and jump chain is?